### PR TITLE
170 bilder mit text verbinden

### PR DESCRIPTION
**Pull request by MariusDoe**

### CODE REVIEW

File: projects/bp-gallery/src/index.tsx [hunk 1]
```diff
@@ -25,11 +25,13 @@ if (matomoUrl) {
...
 }
 
+export const root = document.getElementById('root');
+
 ReactDOM.render(
   <React.StrictMode>
     <BrowserRouter>{renderRoutes(routes)}</BrowserRouter>
   </React.StrictMode>,
-  document.getElementById('root')
+  root
```
Comment (rheaSPK) [inline projects/bp-gallery/src/index.tsx:34]: Um die fehlschlagenden Tests zu fixen hat es bei mir geholfen, diese Datei so zu lassen wie sie ist und stattdessen in den components direkt `document.getElementById('root')` zu nutzen

Comment (MariusDoe) [reply projects/bp-gallery/src/index.tsx:34]: I think waiting for #231 or #221 to be merged will work, too, don't you think?

**Review by olschulz — CHANGES_REQUESTED** [review]
The small PicturePreviews in the ClipboardEditor PictureGrid are still clickable and sometimes result in the entire tab (and dev server) being unresponsive till restart.

I can reproduce this by copying 2 images to the clipboard, then selecting one as a text, and then trying to select the other one as a text by clicking on its preview in the ClipboardEditor.

The curators might never open pictures that way, but I managed to crash it without even trying when I tried to link texts with pictures for the first time

**Comment by olschulz** [comment]
Also getting this warning:
image

File: projects/bp-gallery/src/components/common/picture-gallery/PictureUploadArea.tsx [hunk 2]
```diff
@@ -29,7 +29,7 @@ const PictureUploadArea = ({
   });
   const { t } = useTranslation();
   const { role } = useAuth();
-  const dialog = useContext(DialogContext);
+  const dialog = useDialog();
```
[the remark this answers is not in the document]
Comment (MariusDoe) [reply projects/bp-gallery/src/components/common/picture-gallery/PictureUploadArea.tsx:32]: Strange. `yarn lint` doesn't give that warning for me. I agree that `any` isn't the most beautiful typing here. I just tried to strongly type the whole dialog system but I wasn't able to get something to work. Maybe I will try again some other time, but currently, the linter CI is green so this isn't really high priority, is it?

File: projects/bp-gallery/src/components/views/picture/sidebar/picture-info/PictureInfo.tsx [hunk 3]
```diff
@@ -145,6 +286,106 @@ const PictureInfo = ({
...
+              className='clipboard-button'
+              startIcon={<LinkOff />}
+              variant='contained'
+              onClick={removeAllLinksWithPrompt}
+            >
+              {t('pictureFields.links.removeAll.label')}
+            </Button>
+          )}
+        </PictureInfoField>
+      )}
```
[the remark this answers is not in the document]
Comment (MariusDoe) [reply projects/bp-gallery/src/components/views/picture/sidebar/picture-info/PictureInfo.tsx]: Good idea. Is the new code what you imagined or do you have more requests?

**Comment by MariusDoe** [comment]
> The small PicturePreviews in the ClipboardEditor PictureGrid are still clickable and sometimes result in the entire tab (and dev server) being unresponsive till restart.
> 
> I can reproduce this by copying 2 images to the clipboard, then selecting one as a text, and then trying to select the other one as a text by clicking on its preview in the ClipboardEditor.
> 
> The curators might never open pictures that way, but I managed to crash it without even trying when I tried to link texts with pictures for the first time

I opened #235.

> Also getting this warning: image

Should be fixed in 3afa1d0979fd9787d3eeefafcd602672416c6b37.

File: projects/bp-gallery/src/components/views/picture/sidebar/picture-info/PictureInfo.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (olschulz) [reply projects/bp-gallery/src/components/views/picture/sidebar/picture-info/PictureInfo.tsx]: 👍🏻

File: projects/bp-gallery/src/components/common/picture-gallery/PictureUploadArea.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (olschulz) [reply projects/bp-gallery/src/components/common/picture-gallery/PictureUploadArea.tsx:32]: No, not really high priority :)

**Review by olschulz — APPROVED** [review]

**Review by ole1711 — APPROVED** [review]

**Pull request merged**